aea/agent_loop.py

```diff
@@ -409,21 +252,36 @@ async def _task_process_new_behaviours(self) -> None:
 class SyncAgentLoop(BaseAgentLoop):
     """Synchronous agent loop."""
 
-    def __init__(self, agent: "Agent") -> None:
+    def __init__(self, agent: "Agent", loop: AbstractEventLoop = None):
         """
         Init agent loop.
 
-        :param agent: agent or AEA instance.
+        :param agent: AEA instance
+        :param loop: asyncio loop to use. optional
         """
         super().__init__(agent)
+        self._agent: "AEA" = self._agent
```

Comment:
?

Reply:
cause async event loop uses some methods from AEA and incompatible with agent.
so it's just a type redifinition here.